src/main/java/seedu/address/logic/commands/EditPatientCommand.java

```diff
@@ -0,0 +1,282 @@
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.patient.Diagnosis;
+import seedu.address.model.person.patient.Height;
+import seedu.address.model.person.patient.Patient;
+import seedu.address.model.person.patient.Remark;
+import seedu.address.model.person.patient.Status;
+import seedu.address.model.person.patient.Weight;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Edits the details of an existing patient in the address book.
+ */
+public class EditPatientCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit-ptn";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the patient identified "
+            + "by the index number used in the displayed patients list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_PHONE + "PHONE "
+            + PREFIX_EMAIL + "EMAIL "
+            + PREFIX_HEIGHT + "HEIGHT "
+            + PREFIX_WEIGHT + "WEIGHT "
+            + PREFIX_DIAGNOSIS + "DIAGNOSIS "
+            + PREFIX_STATUS + "STATUS "
+            + PREFIX_REMARK + "REMARK "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_PHONE + "91234567 "
+            + PREFIX_EMAIL + "johndoe@example.com"
+            + PREFIX_HEIGHT + "1.63 ";
+
+
+    public static final String MESSAGE_EDIT_PATIENT_SUCCESS = "Edited Patient: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_PATIENT = "This patient already exists in the address book.";
+
+    private final Index index;
+    private final EditPatientDescriptor editPatientDescriptor;
+
+    /**
+     * @param index of the patient in the filtered patient list to edit
+     * @param editPatientDescriptor details to edit the patient with
+     */
+    public EditPatientCommand(Index index, EditPatientDescriptor editPatientDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editPatientDescriptor);
+
+        this.index = index;
+        this.editPatientDescriptor = new EditPatientDescriptor(editPatientDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Patient> lastShownList = model.getFilteredPatientList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
```

Comment:
Do we want to check if Index is below 0 too?

Reply:
This conditional should handle that scenario